### ISSUE TITLE
[A11Y] amp-carousel Accessibility

### ISSUE THREAD

**Issue by garcialo** [ampproject/amphtml#7975]
Many issues found in a quick review of the amp-carousels on https://ampbyexample.com/components/amp-carousel/

- a) auto updating carousels have no mechanism to Pause, Stop, or Hide the carousel
- b) auto updating carousel doesn't pause on mouse-hover (i.e. when user is trying to interact with content)
- c) keyboard focus / screen reader virtual cursor forced to traverse through the entire carousel content
- d) previous / next buttons not reachable with keyboard / screen reader
- e) no programmatic indication to user that they are going to interact with a carousel component

## How do we reproduce the issue?

a) no Pause, Stop, or Hide for auto updating carousel

1. go to an amp-carousel that auto updates
2. look for a mechanism to pause, stop, or hide the auto-updating content

b) auto updating carousel doesn't pause on hover

1. go to an amp-carousel that auto updates
2. move mouse pointer over the carousel
3. wait for it to auto-update

Note: pausing on hover does not qualify as a method to "pause;" see Understanding WCAG 2.2.2

c) user forced to traverse entire carousel content

1) go to an amp-carousel with multiple slides
2) place keyboard focus before the carousel
3) tab until you get to the content after the carousel

d) previous/next button not keyboard reachable

Repeat steps for c) - previous/next buttons won't aren't reachable

e) no programmatic indication of upcoming carousel

1. go to a page with an amp-carousel
2. enable screen reader
3. navigate to before the carousel
4. traverse the screen reader virtual cursor down the page through the visually presented carousel content

## What browsers are affected?

All browsers.

## Which AMP version is affected?

Current

## Additional Information

Even once these issues are addressed, there will be additional accessibility issues. See the eBay MIND Patterns for a thorough explanation of accessibility best practices, interaction design, etc. for carousel.

https://ebay.gitbooks.io/mindpatterns/content/disclosure/carousel.html

**Comment by aghassemi** [comment]
related: https://github.com/ampproject/amphtml/issues/5487 https://github.com/ampproject/amphtml/issues/5486
